src/main/java/com/saucelabs/bamboo/sod/action/BuildConfigurator.java

```diff
@@ -120,8 +120,7 @@ public void println(String x) {
         SauceTunnelManager sauceTunnelManager = SauceConnectFourManagerSingleton.getSauceConnectFourTunnelManager();
         String options = getResolvedOptions(config.getSauceConnectOptions());
         if (config.useGeneratedTunnelIdentifier()) {
-            String tunnelIdentifier = generateTunnelIdentifier(buildContext.getPlanName());
-            customVariableContext.addCustomData(SODKeys.TEMP_TUNNEL_ID, tunnelIdentifier);
+            String tunnelIdentifier = customVariableContext.getVariables(buildContext).get(SODKeys.TUNNEL_ID);
             options = "--tunnel-identifier " + tunnelIdentifier + " " + options;
```

Comment:
nope unfortunately this does not work `Command line arguments: /Users/moiz/sc-4.3.14-osx/bin/sc -u halkeye -k **** -P 4445 --tunnel-identifier ${bamboo.TUNNEL_IDENTIFIER}`